src/cartesian/Bar.tsx

```diff
@@ -646,6 +647,16 @@ export class Bar extends PureComponent<Props, State> {
     if (hide || !data || !data.length) {
       return (
         <>
+          <SetCartesianGraphicalItem
+            // Bar does not allow setting data directly on the graphical item (why?)
+            data={null}
```

Comment:
hmm, maybe because of the iterations with stacked items? the cases might be too complicated if the data is different?